plugins/module_utils/system_access_users_utils.py

```diff
@@ -809,6 +893,33 @@ def set_user_password(self, user: User) -> None:
         # since "password" is no longer needed, it can be popped
         self._config_maps.pop("password")
 
+    def set_api_keys_secret(self, user: User) -> None:
+        """
+        Sets the API keys for a user, hashing the 'secret' key if not already hashed.
+
+        Args:
+            user (User): The user object containing API keys to be processed.
+
+        Returns:
+            None
+
+        The function iterates over the user's API keys and hashes the 'secret' key using
+        User.generate_hashed_secret if the key name is 'secret' and the value does not already start
+        with "$6$". Other keys and values are left unchanged.
+        """
+
+        user.apikeys = [
+            {
+                key_name: (
+                    User.generate_hashed_secret(secret_value)
+                    if key_name == "secret" and not secret_value.startswith("$6$")
+                    else secret_value
+                )
+                for key_name, secret_value in api_key_dict.items()
+            }
+            for api_key_dict in user.apikeys
+        ]
+
```

Comment:
How are we sure that the secret is also not base64 encoded when hashing it? Could this case occur?

Reply:
outdated, resolving the conversation